crypto/relic_build.sh

```diff
@@ -12,17 +12,48 @@ chmod -R +w "$(pwd)"
 mkdir -p "$DIR/relic/build"
 pushd "$DIR/relic/build"
 
-# Set RELIC config for Flow
 
-#
+# make cmake print its CC interpretation
+CMAKE_FILE="${DIR}/relic/CMakeLists.txt"
+# parameter expansion is not suitable here
+# shellcheck disable=SC2089
+CMAKE_PRINT_CC="message ( STATUS \"CC=\$ENV{CC}\" )"
+# Make the cmake run print its interpretation of CC
+echo "$CMAKE_PRINT_CC" >> "${CMAKE_FILE}"
+
+# Probe cmake's MakeFile generation and extract the CC version
+CMAKE_TEMP=$(mktemp)
+cmake .. > "$CMAKE_TEMP"
+CC_VAL="$(tail -n 5 "$CMAKE_TEMP" | grep -oE -m 1 'CC=.*$')"
+CC_VAL="${CC_VAL:3}"
+
+# de-mangle the CMakeLists file, using a temporary file for BSD compatibility
+sed '$d' ../CMakeLists.txt > "$CMAKE_TEMP"
+mv "$CMAKE_TEMP" ../CMakeLists.txt
```

Comment:
do we need to remove this file at the end (cleanup time? ) or is going to be permanent one

Reply:
The `mv` on line 32 (that you're commenting on) deletes the file already. Besides, `mktemp` creates non-persistent files on most systems.